dcap/types/src/collateral.rs

```diff
@@ -0,0 +1,805 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Collateral for a quote.
+//!
+//! The collateral is additional data that is used to verify the contents of a
+//! [`Quote3`].
+
+extern crate alloc;
+
+use alloc::string::{FromUtf8Error, String};
+use alloc::vec::Vec;
+use mc_sgx_dcap_sys_types::sgx_ql_qve_collateral_t;
+use x509_cert::crl::CertificateList;
+use x509_cert::der::Decode;
+use x509_cert::Certificate;
+
+// Per the comment in `sgx_ql_lib_common.h`
+//
+//      0x00000000: SGX or 0x00000081: TDX
+const SGX_TEE: u32 = 0;
+
+// From the comment in `/etc/sgx_default_qcnl.conf`:
+//
+//      If you use a PCCS service to get the quote verification collateral, you can specify which PCCS API version is to be used.
+//      The legacy 3.0 API will return CRLs in HEX encoded DER format and the sgx_ql_qve_collateral_t.version will be set to 3.0, while
+//      the new 3.1 API will return raw DER format and the sgx_ql_qve_collateral_t.version will be set to 3.1. The pccs_api_version
+//      setting is ignored if collateral_service is set to the Intel PCS. In this case, the pccs_api_version is forced to be 3.1
+//      internally.  Currently, only values of 3.0 and 3.1 are valid.  Note, if you set this to 3.1, the PCCS use to retrieve
+//      verification collateral must support the new 3.1 APIs.
+//
+// This version can be ensured by setting the `pccs_api_version` key of `/etc/sgx_default_qcnl.conf`
+// to `3.1`.
+//
+//      ,"pccs_api_version": "3.1"
+//
+const VERSION_MAJOR: u16 = 3;
+const VERSION_MINOR: u16 = 1;
```

Comment:
Since we have some sway on the enclave host systems I didn't see much need to support the other versions at this time.  If we do want to suggest a follow on PR as this one is already a bit large